doc/sphinx/proof-engine/ltac2.rst

```diff
@@ -27,6 +27,51 @@ especially wherever an advanced tactic language is needed. The previous
 implementation of Ltac, described in the previous chapter, will be referred to
 as Ltac1.
 
+Current limitations include:
+
+- There are a number of tactics that are not yet supported in Ltac2 because
+  the interface OCaml and/or Ltac2 notations haven't been written.  See
+  :ref:`defining_tactics`.
+
+- Missing usability features such as:
+
+  - Printing functions are limited and awkward to use.  Only a few data types are
+    printable.
+  - Deep pattern matching and matching on tuples don't work.
+  - If statements on Ltac2 boolean values
+  - A convenient way to build terms with casts is through the low-level API because the
+    cast type is opaque.  You can use a variation of this code as a workaround:
+
+    .. coqtop:: none
+
+       From Ltac2 Require Import Ltac2.
+
+    .. coqtop:: in
+
+       Ltac2 get_vm_cast () :=
+         match Constr.Unsafe.kind '(I <: True) with
+         | Constr.Unsafe.Cast _ cst _ => cst
+         | _ => Control.throw Not_found
+         end.
+
+- Missing low-level primitives that are convenient for writing automation, such as:
+
+  - An easy way to get the number of constructors of an inductive type.
+    Currently only way to do this is to destruct a variable of the inductive type
+    count the number of goals that result.
+
+  - Building terms with casts currently requires an awkward construction like the
+    following, which also incurs extra overhead to repeat typechecking for each
+    call to `get_vm_cast`.
+
+    .. coqdoc::
+
+       Constr.Unsafe.make (Constr.Unsafe.Cast 'I (get_vm_cast ()) 'True)
```

Comment:
This should probably be moved up and fused with the bullet point above about casts, since there's really only one issue here.